### ISSUE TITLE
Something About Code cleanup:It is recommended to integrate the clang-format formatting tool

### ISSUE THREAD

**Issue by libaineu2004** [codecadwallader/codemaid#717]
### Environment
- Visual Studio version: [example 2015 Community]
- CodeMaid version: [example 10.2]
- Code language: [example C#]
VS2017/C++

### Expected behavior
The official website says:
Code cleanup can run automatically on save, or on demand.
Actions taken on cleanup
1 Remove unused using statements
2 Sort using statements
3 Add unspecified access modifiers
4 Remove empty regions
5 Add blank line padding
6 Remove blank lines next to braces
**7 Run Visual Studio formatting**

I think the current formatting method is relatively fixed and not flexible enough.
I suggest integrating the clang-format formatting tool together. There is a corresponding interface that can be set up, which allows users to choose which style to adopt, such as LLVM, Google, Microsoft, etc., and can also customize files.

**Comment by codecadwallader** [comment]
Thanks for the suggestion.  You may also want to check out #346 for an approach to integrate it as a third party tool.

**Comment by libaineu2004** [comment]
> Thanks for the suggestion. You may also want to check out #346 for an approach to integrate it as a third party tool.

Thank you. I saw the topic # 346. We very much hope that CodeMaid can integrate the clang-format tool itself and provide visual tabs instead of using other external tools of Visual Studio.
Looking forward to completion soon, blessings!

**Comment by libaineu2004** [comment]
Is there a development plan for the integrated clang-format function? When will it be completed?

**Comment by codecadwallader** [comment]
No, there is not currently a development plan for this feature.  As an open source project it is open for anyone who is interested in pursuing its implementation.
